docs/scoping.md

```diff
@@ -228,3 +228,43 @@ references to be resolved until all references are resolved or no progress
 regarding the resolution is observed. In the latter case an error is raised. The
 control flow responsibility of the resolution process is allocated to the
 `model.py` module.
+
+
+### Using the scope provider to modify a model
+
+Scoping takes place after the model is completely parsed. Thus, 
+while resolving references you can rely on the assumption that all
+model elements already exist.
+
+However, **if your scope provider creates (invents) model elements**, it may
+happen, that also model elements to be referenced do not yet exist. This may
+happen, if your meta model allows to define a model element by referencing
+it (like [PlantUML](http://plantuml.com/) is doing for, e.g., classes).
+If you then require to reference these model elements "defined by
+a reference", you must take into account that these elements
+may have not yet been created. This can be achieved in the same 
+way as handling unresolved references in a scope provider (with the 
+```Postponed``` mechanism). This use case was motivated by 
+[#167](https://github.com/textX/textX/issues/167).
+
+An example of such a meta model is given in 
+[tests/test_model_modification_through_scoping.py](https://github.com/textX/textX/blob/master/tests/functional/test_scoping/test_model_modification_through_scoping.py):
+Here you can 
+ * **define** Persons explicitly (existence) and
```

Comment:
After `Here you can` and additional newline is needed to render the bullets list properly.